frontend/src/modules/Metadata_Forms/components/MetadataFormFields.js

```diff
@@ -0,0 +1,424 @@
+    setLocalFields([...localFields]);
+  };
+  const addField = () => {
+    localFields.push({
+      name: 'New Field',
+      required: false,
+      metadataFormUri: formUri,
+      type: fieldTypeOptions[0].value,
+      possibleValues: [],
+      deleted: false
+    });
+    setLocalFields([...localFields]);
+  };
+
+  return (
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableCell sx={{ width: '20px' }}>Required</TableCell>
+          <TableCell>Name</TableCell>
+          <TableCell>Type</TableCell>
+          <TableCell sx={{ width: '30%' }}>Description</TableCell>
+          <TableCell sx={{ width: '15%' }}>
+            Values (any, if not specified)
+          </TableCell>
+          <TableCell sx={{ width: '20px' }}>
+            <Button
+              color="primary"
+              startIcon={<SaveIcon size={15} />}
+              sx={{ mt: 1 }}
+              onClick={() => {
+                saveChanges(localFields);
+              }}
+              type="button"
+              variant="outlined"
+            >
+              Save
+            </Button>
+          </TableCell>
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        <TableRow>
+          <TableCell colSpan={6}>
+            <Button
+              color="primary"
+              startIcon={<PlusIcon size={15} />}
+              sx={{ mt: 1 }}
+              onClick={addField}
+              type="button"
+            >
+              Add field
+            </Button>
+          </TableCell>
+        </TableRow>
+        {localFields.length === 0 ? (
+          <TableRow>
+            <TableCell colSpan={6} align="center">
+              No fields found
+            </TableCell>
+          </TableRow>
+        ) : (
+          localFields.map((field, index) => (
+            <TableRow
+              sx={{
+                backgroundColor: field.deleted ? 'whitesmoke' : 'white'
+              }}
+            >
+              <TableCell>
+                <Checkbox
+                  defaultChecked={field.required}
+                  disabled={field.deleted}
+                  onChange={(event) => {
+                    updateField(index, 'required', event.target.value === 'on');
+                  }}
+                />
+              </TableCell>
+              <TableCell>
+                <TextField
+                  disabled={field.deleted}
+                  defaultValue={field.name}
+                  onKeyUp={(event) => {
+                    updateField(index, 'name', event.target.value);
+                  }}
+                  sx={{ width: '100%' }}
+                />
+              </TableCell>
+              <TableCell>
+                <Autocomplete
+                  disablePortal
+                  disabled={field.deleted}
+                  options={fieldTypeOptions.map((option) => option.value)}
+                  defaultValue={field.type}
+                  onKeyUp={(event, value) => {
```

Comment:
for `AutoComplete` this should be `onChange` rather that `onKeyUp` otherwise the Type remains defaulted to `String` in my testing

Reply:
Good catch

Reply:
[72e025d](https://github.com/data-dot-all/dataall/pull/1455/commits/72e025dbe76b1818d5ca0d71bc52bb034d1aef1b)